scripts/popup.js

```diff
@@ -205,14 +205,30 @@ function borrow_books() {
     tabId = tabs[0].id
     chrome.browserAction.getBadgeText({ tabId: tabId }, function (result) {
       if (result.includes('B') && url.includes('www.amazon') && url.includes('/dp/')) {
-        get_amazonbooks(url).then(response => {
-          if (response['metadata'] && response['metadata']['identifier-access']) {
-            let details_url = response['metadata']['identifier-access']
+        chrome.storage.sync.get(['tab_url', 'detail_url'], function (result) { 
+          let stored_url = result.tab_url
+          let detail_url = result.detail_url
+          // Checking if the tab url is the same as the last stored one
+          console.log(stored_url, url, detail_url)
```

Comment:
Please try not to include these statements in your final code.

Reply:
Yes, I am adding these for your debugging purpose. I will delete this once you have approved my PR